database/src/main/scala/cromwell/database/slick/BackendKVStoreSlickDatabase.scala

```diff
@@ -11,12 +11,20 @@ trait BackendKVStoreSlickDatabase extends BackendKVStoreSqlDatabase {
   import dataAccess.driver.api._
 
 
-  override def addBackendKVStoreEntry(entry: BackendKVStoreEntry)(implicit ec: ExecutionContext): Future[Unit] = {
+  override def addBackendJobKeyValuePair(workflowID: String, callFqn: String, callIndex: Int, callAttempt: Int, storeKey: String, storeValue: String)(implicit ec: ExecutionContext): Future[Unit] = {
+
+    val entry = BackendKVStoreEntry(workflowID.toString,
+                              callFqn,
+                              callIndex,
+                              callAttempt,
+                              storeKey,
+                              storeValue)
+
     val action = dataAccess.backendKVStoreAutoInc += entry
```

Comment:
While this action is fine for inserting, I believe the key value store should be update-or-inserting, aka upserting.

```scala
val entry = BackendKVStoreEntry(workflowUuid,
  callFqn,
  callIndex,
  callAttempt,
  storeKey,
  storeValue)
val action = if (useSlickUpserts) {
  // When using slick based upserts, as long as a unique constraint has been created on the columns belonging
  // to the composite-unique-key, then the rest of the values will be upserted appropriately.
  for {
    _ <- dataAccess.backendKVStoreAutoInc.insertOrUpdate(entry)
  } yield () // <-- maps the result to a DBIO[Unit]
} else {
  // If slick can't upsert for us, then we must within one transaction try to update the row first, and if no
  // row was found then we should insert one.
  // NOTE: Because this upsert has a non-zero chance of failing due to race conditions, the caller
  // should always call this upsert using a `cromwell.util.DatabaseUtil.withRetry()`.
  for {
    updateCount <- dataAccess.
      backendJobValueByBackendJobKey(workflowUuid, callFqn, callIndex, callAttempt, storeKey).
      update(storeValue)
    _ <- updateCount match {
      case 0 => dataAccess.backendKVStoreAutoInc += entry
      case _ => assertUpdateCount("addBackendJobKeyValuePair", updateCount, 1)
    }
  } yield () // <-- maps the result to a DBIO[Unit]
}
runTransaction(action)
```